### PR TITLE
Ops 675/task 1060 form validation

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

Adds form validation with the Vest library.  Adds POC with the Project Create form.

## Issue

#675 and #1060

## How to test

Goto Create a new Project and test to see if all form fields must be entered and select must be selected. Submit button will be disabled until all form validation is complete.

## Screenshots

image

## Links

[Vest docs](https://vestjs.dev/)

### CODE REVIEW

File: frontend/src/pages/projects/CreateProject.jsx [hunk 1]
```diff
@@ -1,36 +1,49 @@
...
+        invalid: "usa-form-group--error",
+        valid: "success",
+        warning: "warning",
+    });
+
+    // prepare data for submission by removing the type field
+    const editedProject = {
+        ...project,
+    };
+    delete editedProject.type;
```
Comment (fpigeonjr) [inline frontend/src/pages/projects/CreateProject.jsx:45]: had to remove the project type prop or it wouldn't POST

File: frontend/src/pages/projects/CreateProject.jsx [hunk 2]
```diff
@@ -94,59 +107,50 @@ export const CreateProject = () => {
...
             <div className="grid-row flex-justify-end margin-top-8">
                 <button id="cancel" className="usa-button usa-button--unstyled margin-right-2" onClick={handleCancel}>
                     Cancel
                 </button>
-                <button id="submit" className="usa-button" onClick={() => addResearchProject(project)}>
+                <button
+                    id="submit"
+                    className="usa-button"
+                    onClick={() => addResearchProject(editedProject)}
+                    disabled={!res.isValid()}
```
Comment (fpigeonjr) [inline frontend/src/pages/projects/CreateProject.jsx:152]: disabled the submit button if all validation rules haven't passed

File: frontend/src/pages/projects/suite.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { create, test, enforce, only } from "vest";
+
+const suite = create((data = {}, fieldName) => {
+    only(fieldName);
+
+    test("short_title", "This is required information", () => {
+        enforce(data.short_title).isNotBlank();
+    });
+    test("short_title", "Nickname must be 3 letters", () => {
```
Comment (fpigeonjr) [inline frontend/src/pages/projects/suite.js:9]: made this up, not sure if needed

**Review by tdonaworth — APPROVED** [review]

File: frontend/src/pages/projects/suite.js (hunk 3 above)
Comment (tdonaworth) [reply frontend/src/pages/projects/suite.js:9]: Seems valid, might want to reflect it in a .feature rule though.

**Pull request merged**